src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -0,0 +1,186 @@
+package mesosphere.marathon.core.launchqueue.impl
+
+import akka.actor.SupervisorStrategy.Stop
+import akka.actor.{
+  PoisonPill,
+  Terminated,
+  Actor,
+  ActorLogging,
+  ActorRef,
+  OneForOneStrategy,
+  Props,
+  SupervisorStrategy
+}
+import akka.event.LoggingReceive
+import akka.pattern.{ ask, pipe }
+import akka.util.Timeout
+import mesosphere.marathon.core.launchqueue.LaunchQueue
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+import LaunchQueue.QueuedTaskCount
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+private[launchqueue] object LaunchQueueActor {
+  def props(appActorProps: (AppDefinition, Int) => Props): Props = {
+    Props(new LaunchQueueActor(appActorProps))
+  }
+
+  case class FullCount(appId: PathId)
+}
+
+/**
+  * An actor-based implementation of the [[LaunchQueue]] interface.
+  *
+  * The methods of that interface are translated to messages in the [[ActorLaunchQueue]] implementation.
+  */
+private[impl] class LaunchQueueActor(appActorProps: (AppDefinition, Int) => Props) extends Actor with ActorLogging {
```

Comment:
This actor doesn't see start stop messages. It is simply started/stopped. If it gets stopped, it state gets discarded automatically.